mos-platform/common/c/malloc.cc

```diff
@@ -16,6 +16,20 @@
 
 extern char __heap_start;
 extern char __heap_default_limit;
+extern char __stack_reserve_size;
+
+// Get current soft stack pointer value
+__asm__ (
+".text\n"
+".global __get_current_stack_top\n"
```

Comment:
We can avoid making this a global symbol at all by using inline assembly in a static function. This would also allow the compiler to inline this, which would usually be beneficial.